apps/ensadmin/src/app/gql/api/route.ts

```diff
@@ -0,0 +1,286 @@
+  options: GetQueryGeneratorClientOptions,
+): Promise<QueryGeneratorClient> {
+  let client = clients.get(options.gqlApiUrl.toString());
+
+  if (!client) {
+    // create the client if it doesn't exist yet
+    client = new GQLPTClient({
+      url: options.gqlApiUrl.toString(),
+      adapter: new AdapterAnthropic({
+        apiKey: options.llmApiKey,
+        model: Model.Claude35Sonnet,
+        systemPrompt: SYSTEM_PROMPT,
+        maxTokensPerMessage: MAX_TOKENS_PER_MESSAGE,
+      }),
+    });
+
+    try {
+      // ensure the client is connected
+      await client.connect();
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : "Unknown error";
+      throw new Error(`Failed to connect to the LLM: ${errorMessage}`);
+    }
+
+    // store the client in the map for future use
+    clients.set(options.gqlApiUrl.toString(), client);
+  }
+
+  return client;
+}
+
+interface ParseRequestOptions {
+  maybeGqlApiUrl: string | null;
+  maybePrompt: string | null;
+}
+
+/**
+ * Parse the request URL to get the prompt and GQL API URL.
+ *
+ * @param requestUrl The request URL
+ * @returns The prompt and GQL API URL
+ * @throws {Error} If the prompt or GQL API URL was not provided
+ */
+getQueryGeneratorClient.parseRequest = function parseQueryGeneratorClientRequest(
+  options: ParseRequestOptions,
+): GenerateQueryDto {
+  return GenerateQueryDto.tryParse(options.maybePrompt, options.maybeGqlApiUrl);
+};
+
+/**
+ * DTO for the generateQueryAndVariables request.
+ */
+class GenerateQueryDto {
+  private constructor(
+    public readonly prompt: string,
+    public readonly gqlApiUrl: URL,
+  ) {}
+
+  static tryParse(maybePrompt: string | null, maybeGqlApiUrl: string | null) {
+    const prompt = GenerateQueryDto.parsePrompt(maybePrompt);
+    const gqlApiUrl = GenerateQueryDto.parseGqlApiUrl(maybeGqlApiUrl);
+
+    return new GenerateQueryDto(prompt, gqlApiUrl);
+  }
+
+  static parsePrompt(maybePrompt: string | null) {
+    if (!maybePrompt) {
+      throw new Error("Prompt is required");
+    }
+
+    return maybePrompt;
+  }
+
+  static parseGqlApiUrl(maybeGqlApiUrl: string | null) {
+    if (!maybeGqlApiUrl) {
+      throw new Error("URL is required");
+    }
+
+    try {
+      return new URL(maybeGqlApiUrl);
+    } catch (error) {
+      throw new Error("Invalid URL");
+    }
+  }
+}
+
+interface AdapterAnthropicOptions extends ClientOptions {
+  /** The Anthropic model to use */
+  model: Model;
+
+  /** The system prompt to use */
+  systemPrompt: string;
+
+  /** The max tokens to use per message */
+  maxTokensPerMessage: number;
+}
+
+enum Model {
+  Claude35Sonnet = "claude-3-5-sonnet-20241022",
+  Claude37Sonnet = "claude-3-7-sonnet-20250219",
+}
+
+/**
+ * Adapter for Anthropic with selectable model and system prompt.
+ *
+ * Based on https://github.com/rocket-connect/gqlpt/blob/18af9c9/packages/adapter-anthropic/src/index.ts
+ */
+class AdapterAnthropic extends Adapter {
```

Comment:
Our anthropic Adapter is not passing through the options and hardcoded on a specific model - sorry about that, I made something on our end to track that: 

* https://github.com/rocket-connect/gqlpt/issues/174